services/scheduler/scheduler.go

```diff
@@ -157,43 +118,14 @@ func main() {
 		return err
 	})
 
-	// scheduler goroutine
-	g.Go(func() error {
-		client := pb.NewContextBoxServiceClient(cc)
-		prevState := cc.GetState()
-		group := sync.WaitGroup{}
-
-		worker.NewWorker(
-			ctx,
-			10*time.Second,
-			func() error {
-				if cc.GetState() == connectivity.Ready {
-					if prevState != connectivity.Ready {
-						log.Info().Msgf("Connection to Context-box is now ready")
-					}
-					prevState = connectivity.Ready
-				} else {
-					log.Warn().Msgf("Connection to Context-box is not ready yet")
-					log.Debug().Msgf("Connection to Context-box is %s, waiting for the service to be reachable", cc.GetState().String())
-
-					prevState = cc.GetState()
-					cc.Connect() // try connecting to the service.
-
-					return nil
-				}
-				return configProcessor(client, &group)
-			},
-			worker.ErrorLogger,
-		).Run()
-
-		log.Info().Msg("Scheduler stopped checking for new configs")
-		log.Info().Msgf("Waiting for already started configs to finish processing")
-
-		group.Wait()
-		log.Debug().Msgf("All spawned go-routines finished")
-
-		return nil
-	})
+	log.Info().Msgf("Stopping Scheduler: %v", errGroup.Wait())
+}
 
-	log.Info().Msgf("Stopping Scheduler: %v", g.Wait())
+// healthCheck function is used for querying readiness of the pod running this microservice
+func healthCheck(usecases *usecases.Usecases) error {
```

Comment:
Would it make sense to move cbox connection check into the health check function @Despire?

Reply:
It would makes sense, though there would be a question what to do about the worker. It pulls a config every few seconds and it would wouldn't be able to connect to cbox that it would just flood the logs with these error messages instead of printing cbox not available.